loom/engine/bindings/loom/lmModestMaps.cpp

```diff
@@ -0,0 +1,311 @@
+    static float gridTRy;
+    static float gridBLx;
+    static float gridBLy;
+    static float gridCx;
+    static float gridCy;
+
+
+    static const char *tileKey(int col, int row, int zoom)
+    {
+        static char key[256];
+        sprintf(key, "%c:%i:%i", (char)(((int)'a')+zoom), col, row);
+        return (const char *)key;
+    }
+ 
+    static const char *prepParentLoad(int col, int row, int zoom, int parentZoom)
+    {
+        //NOTE_TEC: zoomDiff should always be +ve
+        int zoomDiff = zoom - parentZoom;
+        if(zoomDiff <= 0)
+        {
+            parentLoadCol = col;
+            parentLoadRow = row;
+            parentLoadZoom = zoom;
+        }
+        else
+        {
+            float invScaleFactor = 1.0f / (float)(1 << zoomDiff);
+            parentLoadCol = (int) floor((float)col * invScaleFactor); 
+            parentLoadRow = (int) floor((float)row * invScaleFactor);
+            parentLoadZoom = parentZoom;
+        }
+        return tileKey(parentLoadCol, parentLoadRow, parentLoadZoom);
+    }
+
+    static void setLastCoordinate(float col,
+                                    float row,
+                                    float zoom,
+                                    float zoomLevel,
+                                    float invTileWidth,
+                                    Matrix *worldMatrix,
+                                    DisplayObject *context,
+                                    DisplayObject *object)
+    {
+        // this is basically the same as coord.zoomTo, but doesn't make a new Coordinate:
+        float zoomFactor = pow(2, zoomLevel - zoom) * invTileWidth;
+        float zoomedColumn = col * zoomFactor;
+        float zoomedRow = row * zoomFactor;
+                    
+        worldMatrix->transformCoordInternal(zoomedColumn, zoomedRow, &lastCoordinateX, &lastCoordinateY);
+
+        //transform into correct space if necessary
+        if ((context != NULL) && (context != object))
+        {
+            localToGlobal((DisplayObject *)(object->parent), &lastCoordinateX, &lastCoordinateY);
+            globalToLocal(context, &lastCoordinateX, &lastCoordinateY);
+        }
+    } 
+
+
+    static void setGridCoordinates(Matrix *invMatrix, float mapWidth, float mapHeight, float mapScale)
+    {
+        const float LN2 = 0.6931471805599453f;
+        gridZoom = log(mapScale) / LN2;
+        invMatrix->transformCoordInternal(0.0f, 0.0f, &gridTLy, &gridTLx);
+        invMatrix->transformCoordInternal(mapWidth, mapHeight, &gridBRy, &gridBRx);
+        invMatrix->transformCoordInternal(mapWidth, 0.0f, &gridTRy, &gridTRx);
+        invMatrix->transformCoordInternal(0.0f, mapHeight, &gridBLy, &gridBLx);
+        invMatrix->transformCoordInternal(mapWidth * 0.5f, mapHeight * 0.5f, &gridCy, &gridCx);
+    }
+
+
+    static Matrix *getGridInverseMatrix(Matrix *worldMatrix, float tileWidth, float tileHeight, float mapScale)
+    {
+        Matrix *invMatrix = new Matrix();
```

Comment:
Umm. Yeah I'm not sure on semantics. Can we make it pass by ref so that we don't allocate on every call?